File: 350-IntersectionOfTwoArraysIi/350-IntersectionOfTwoArraysIi.js
// Last updated: 4/29/2026, 10:30:52 AM
/**
 * @param {number[]} nums1
 * @param {number[]} nums2
 * @return {number[]}
 */
var intersect = function(nums1, nums2) {
    const map = {};
    for(const num1 of nums1) {
        if(map[num1] === undefined) {
            map[num1] = 1;
        } else {
            map[num1]++;
        }
    }
    console.log(map);
    const res = [];
    for(const num2 of nums2) {
        if(map[num2] > 0) {
            res.push(num2);
            map[num2]--;
        }
    }
    return res;
};

File: 350-IntersectionOfTwoArraysIi/350-IntersectionOfTwoArraysIi.py
# Last updated: 4/29/2026, 10:30:57 AM
class Solution:
    def intersect(self, nums1: List[int], nums2: List[int]) -> List[int]:
        from collections import defaultdict
        dict = defaultdict(lambda: 0)
        for num1 in nums1:
            dict[num1] += 1
        print(dict)
        res = []
        for num2 in nums2:
            if dict[num2] > 0:
                dict[num2] -= 1
                res.append(num2)
        return res